docs/enhancements/clusterpool-inventory.md

```diff
@@ -0,0 +1,175 @@
+    ```json
+    [
+      { "op": "replace", "path": "metadata/name", "value": "foo" }
+    ]
+    ```
+- Add the name of InstallConfigPatch CR to`clusterPool.spec.inventory.installConfigPatches` list. For InstallConfigPatch with a name `foo-install-config-patch` the clusterpool should be configured as follows
+    ```yaml
+    spec:
+      inventory:
+        installConfigPatches:
+          - name: foo-install-config-patch
+    ```
+  
+### Validation
+Webhook validation will ensure that for a clusterpool
+- if `spec.inventory.installConfigPatchs` is specified, it is not an empty list.
+- if `spec.inventory.installConfigPatchs` is specified, `spec.maxSize` is equal to or less than length of list `spec.inventory.installConfigPatchs`.
+- if `spec.inventory.installConfigPatchs` is specified, `spec.size` is equal to or less than length of list `spec.inventory.installConfigPatchs`.
+
+### `Size` and `MaxSize`
+If `Inventory` is used, `ClusterPool.Spec.Size` and `.MaxSize` are implicitly constrained to the length of the list of nonempty `spec.inventory.installConfigPatchs`.
+Setting either/both to a smaller number still works as you would expect.
+
+### Pool Version
+To make [adding](#adding-an-inventory) and [deleting](#deleting-the-inventory) `Inventory` work sanely, we will adjust the computation of the pool version used for [stale CD detection/replacement](https://issues.redhat.com/browse/HIVE-1058) as follows:
+- When `Inventory` is present, append an arbitrary fixed string before the [final hash operation](https://github.com/openshift/hive/blob/0b9229b91e6f8a3c2bf095efbdf017226e69d026/pkg/controller/clusterpool/clusterpool_controller.go#L479).
+  (We don't want to recalculate the hash any time the inventory changes, as this would treat *all* existing unclaimed CDs as stale and replace them.)
+- When `Inventory` is absent, compute the pool version as before.
+  This ensures the version does not change (triggering replacement of all unclaimed CDs) for existing pools when hive is upgraded to include this feature.
+
+### Handling Inventory Updates
+#### Adding An Inventory
+Adding an `Inventory` to a ClusterPool which previously didn't have one will cause the controller to [recompute the pool version](#pool-version), rendering all existing unclaimed clusters stale, causing them to be replaced gradually.
+
+#### Adding An Entry to the Inventory
+If `MaxSize` is unset or less than/equal to the length of the inventory, and `Size` allows, adding a new entry will cause a new ClusterDeployment to be added to the pool.
+If the pool is already at `[Max]Size` there is no immediate effect.
+
+#### Removing An Entry to the Inventory
+- If the entry is unused (no pool CD with that name exists), this is a no-op.
+- If an _unclaimed_ CD exists with that name, we delete it.
+  The controller will replace it, assuming an unused entry is available.
+- If a _claimed_ CD exists with that name, no-op.
```

Comment:
I feel like someone is inevitably going to ask "i deleted from the inventory. how do i know if it is presently in use?"

Reply:
We can just go ahead and check the status on the CRD